### PR TITLE
added validation for source and target

### PR DESCRIPTION
**Pull request by AshwinKul28**
- Fixed a bug where the user can create a reverse proxy even if the host or target is empty

Related issue: https://github.com/vertex-center/vertex/issues/54

### CODE REVIEW

File: apps/reverseproxy/handler/proxy.go [hunk 1]
```diff
@@ -37,6 +38,24 @@ func (r *ProxyHandler) AddRedirect(c *router.Context) {
...
+	}
+
+	if body.Target == config.EmptyStr {
+		c.Abort(router.Error{
+			Code:           types2.ErrCodeInvalidAddRedirectUuid,
+			PublicMessage:  fmt.Sprintf("Failed to add redirect as target is empty."),
+			PrivateMessage: err.Error(),
+		})
+		return
+	}
```
[the remark this answers is not in the document]
Comment (AshwinKul28) [reply apps/reverseproxy/handler/proxy.go]: Hi @quentinguidee Thanks for this. I got confused about whether to add this validation only for HTTP handler or in the core logic. Updated the code and tested on the local dev environment as well. Please check.

File: config/config.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ type Mode string
 const (
 	ProductionMode Mode = "production"
 	DebugMode      Mode = "debug"
+	EmptyStr            = ""
```
[the remark this answers is not in the document]
Comment (AshwinKul28) [reply config/config.go]: Removed this line. As I was thinking to have a constant for empty string.

**Comment by quentinguidee** [comment]
Sorry for the confusion; I think the documentation should be more explicit about this.

In fact, both versions were half-right. This new version check the errors at the correct place, but doesn't returns to the client why the request has failed.
- The service should return, for example, an `ErrSourceInvalid` or `ErrTargetInvalid` depending on the error;
- Then, the handler should receive the error given by the service (eg. `if errors.Is(err, ErrSourceInvalid) {...}`) and returns an HTTP response corresponding to the error.

Take for example the feature to delete a container: https://github.com/vertex-center/vertex/blob/dev/apps/containers/handler/container.go#L120:L135. If the service says that the service is still running, the handler returns a response that is adapted to that.

So: the service makes the verification and returns why it fails, and the http handler generates an HTTP response for this error.

But appart from this I think you got it!

For the empty string, this is optional for this PR but you can also use `strings.TrimSpace()` to also invalidate sources/targets like `"    "` (with only spaces).

**Comment by quentinguidee** [comment]
(Also don't hesitate to say what should be improved in the documentation, the doc is pretty new!)

**Review by quentinguidee — APPROVED** [review]

**Comment by AshwinKul28** [comment]
Thanks @quentinguidee for the explanation. Actually, my both solutions may be an amalgam of what actually this repo expects. 

One more suggestion, in the future maybe you can add pre-commit hooks that will have go-linter inbuilt that will make the code aesthetically correct.

**Comment by AshwinKul28** [comment]
fixing the lint errors.

**Comment by quentinguidee** [comment]
Looks good to me! Thank you for your contribution 🚀

**Pull request merged**